vertical-pod-autoscaler/pkg/recommender/input/metrics/metrics_source.go

```diff
@@ -164,12 +286,26 @@ func (s podMetricsSource) List(ctx context.Context, namespace string, opts v1.Li
 		return nil, err
 	}
 
-	if s.m3Url == "" {
+	if s.m3Url == nil {
 		klog.Info("No M3 URL provided - skipping custom pod usage metrics")
 		return podsMetrics, nil
 	}
 
-	return s.withM3CustomMetrics(podsMetrics)
+	customQueries := []nsQuery{
+		{
+			f:                  getRSSQuery,
+			resource:           k8sapiv1.ResourceName(model.ResourceRSS),
+			containerNameLabel: "container_name",
+			podNameLabel:       "pod_name",
+		},
+		{
+			f:                  getJVMHeapCommittedQuery,
+			resource:           k8sapiv1.ResourceName(model.ResourceJVMHeapCommitted),
+			containerNameLabel: "kubernetes_container_name",
+			podNameLabel:       "kubernetes_pod_name",
+		},
+	}
+	return s.withCustomResourceMetrics(podsMetrics, customQueries)
```

Comment:
We should try calling the `podsMetrics, err := podMetricsInterface.List(ctx, opts)` and custom metrics concurrently. This way the latency of fetching metrics would probably be cut in half.

Reply:
I suspect the slowness here is due to the JSON unmarshalling of the metrics response. I'll leave a TODO to use a prom client instead

Reply:
>I suspect the slowness here is due to the JSON unmarshalling of the metrics response.

In this case, would increasing CPU Limit help :)